x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsQueryExec.java

```diff
@@ -57,13 +59,14 @@ public FieldSortBuilder fieldSortBuilder() {
         }
     }
 
-    public EsQueryExec(Source source, EsIndex index, QueryBuilder query) {
-        this(source, index, List.of(new FieldAttribute(source, DOC_ID_FIELD.getName(), DOC_ID_FIELD)), query, null, null, null);
+    public EsQueryExec(Source source, EsIndex index, IndexMode indexMode, QueryBuilder query) {
+        this(source, index, indexMode, sourceFields(source, indexMode), query, null, null, null);
     }
 
     public EsQueryExec(
         Source source,
         EsIndex index,
+        IndexMode indexMode,
```

Comment:
Yeah, I've tried that before, but the problem is that this EsIndex is used in the QL module and many other places. Let me replicate it in ESQL and add the index mode to the new EsIndex.

Reply:
Hope #108773 will make things simpler.

Reply:
I gave this a try, but we create EsIndex inside IndexResolver, which depends on the field-caps responses rather than the command. I will revisit this again after we have index mode in field-caps.